src/bentoml/_internal/monitoring/api.py

```diff
@@ -30,257 +23,126 @@
-        raise NotImplementedError()
-
-    def export_schema(self) -> JSONSerializable:
-        raise NotImplementedError()
-
-    def export_data(self) -> JSONSerializable:
-        raise NotImplementedError()
-
-    def log(self, data: DT, name: str, role: str, data_type: str) -> None:
-        raise NotImplementedError()
-
-    def log_batch(
-        self,
-        data_batch: t.Iterable[DT],
-        name: str,
-        role: str,
-        data_type: str,
-    ) -> None:
-        raise NotImplementedError()
-
-    def log_table(
-        self,
-        data: t.Iterable[t.Iterable[DT]],
-        schema: dict[str, str],
-    ) -> None:
-        raise NotImplementedError()
-
-
-class NoOpMonitor(MonitorBase[t.Any]):
-    def __init__(self, name: str, **kwargs: t.Any) -> None:
-        pass
-
-    def start_record(self) -> None:
-        pass
-
-    def stop_record(self) -> None:
-        pass
-
-    def export_schema(self) -> JSONSerializable:
-        pass
-
-    def export_data(self) -> JSONSerializable:
-        pass
-
-    def log(self, data: t.Any, name: str, role: str, data_type: str) -> None:
-        pass
-
-    def log_batch(
-        self,
-        data_batch: t.Iterable[t.Any],
-        name: str,
-        role: str,
-        data_type: str,
-    ) -> None:
-        pass
-
-    def log_table(
-        self,
-        data: t.Iterable[t.Iterable[t.Any]],
-        schema: dict[str, str],
-    ) -> None:
-        pass
 
-
-DEFAULT_CONFIG_YAML = """
-version: 1
-incremental: false
-loggers:
-  bentoml_monitor_data:
-    level: INFO
-    handlers: [bentoml_monitor_data]
-  bentoml_monitor_schema:
-    level: INFO
-    handlers: [bentoml_monitor_schema]
-handlers:
-  bentoml_monitor_data:
-    class: logging.handlers.TimedRotatingFileHandler
-    level: INFO
-    formatter: bentoml_json
-    filename: "{data_filename}"
-    when: "D"
-  bentoml_monitor_schema:
-    class: logging.handlers.RotatingFileHandler
-    level: INFO
-    formatter: bentoml_json
-    filename: "{schema_filename}"
-formatters:
-  bentoml_json:
-    class: pythonjsonlogger.jsonlogger.JsonFormatter
-    format: " "
-"""
-
-
-class DefaultMonitor(MonitorBase["JSONSerializable"]):
+class MonitorBase(t.Generic[DT]):
     """
-    The default monitor implementation. It uses a logger to log data and schema, and will
-    write monitor data to rotating files. The schema is logged as a JSON object, and the
-    data is logged as a JSON array.
+    The base monitor class. All monitors should inherit from this class.
+    Subclasses should implement the following methods:
+    - export_schema
+    - export_data
+    to export the schema and data to the desired format.
     """
 
-    PRESERVED_COLUMNS = (COLUMN_TIME, COLUMN_RID) = ("timestamp", "request_id")
+    PRESERVED_COLUMNS: tuple[str, ...] = ()
 
     def __init__(
         self,
         name: str,
-        log_path: str,
-        log_config_file: str | None = None,
+        **_: t.Any,
```

Comment:
Lets' not do this, it should be using __init_subclass__?

Reply:
why?

Reply:
isn't _ the same?